packages/flutter_tools/lib/src/ios/ios_workflow.dart

```diff
@@ -33,6 +32,19 @@ class IOSWorkflow extends DoctorValidator implements Workflow {
 
   bool get hasIDeviceId => exitsHappy(<String>['idevice_id', '-h']);
 
+  bool get hasWorkingLibimobiledevice {
+    // Verify that libimobiledevice tools are installed.
+    if (!hasIDeviceId)
+      return false;
+
+    // If a device is attached, verify that we can get its name.
+    final String result = runSync(<String>['idevice_id', '-l']);
```

Comment:
The entire library uses sync methods right now; will follow up with a patch that migrates everything over to the async equivalents.